CVE-2021-3156/ubuntu.yml

```diff
@@ -0,0 +1,91 @@
+---
+- name: Install and test CVE-2021-3156 (sudo privilege escalation)
+  hosts: ubuntu
+  become: yes
+  vars:
+    sudo_version: "1.8.31"
+    install_dir: "/opt/sudo"
+    source_dir: "/opt/sudo/sudo-{{ sudo_version }}"
+    exploit_dir: "/opt/exploit"
+
+  tasks:
+    # Ensure build dependencies are installed
+    # - name: Install build dependencies
+    #   apt:
+    #     name:
+    #       - build-essential
+    #       - libssl-dev
+    #       - libpam0g-dev
+    #       - zlib1g-dev
+    #     state: present
+
+    # Create the installation directory for sudo
+    - name: Create sudo installation directory
+      file:
+        path: "{{ install_dir }}"
+        state: directory
+
+    # Copy the vulnerable sudo tar file to the target VM
+    - name: Copy vulnerable sudo tar file
+      copy:
+        src: "./data/sudo-{{ sudo_version }}.tar.gz"
+        dest: "{{ install_dir }}/sudo-{{ sudo_version }}.tar.gz"
+
+    # Extract the tar file
+    - name: Extract vulnerable sudo tar file
+      unarchive:
+        src: "{{ install_dir }}/sudo-{{ sudo_version }}.tar.gz"
+        dest: "{{ install_dir }}"
+        remote_src: yes
+
+    # Compile and install the vulnerable sudo version
+    - name: Compile and install vulnerable sudo
```

Comment:
It fails:

> 2024-11-21 12:00:20,277 - INFO - [ubuntu] TASK [Compile and install vulnerable sudo] *************************************
> 2024-11-21 12:00:21,582 - INFO - [ubuntu] fatal: [ubuntu]: FAILED! => {"changed": true, "cmd": "./configure --prefix=/usr --enable-static\nmake\nmake install\n", "delta": "0:00:00.492920", "end": "2024-11-21 11:00:19.482098", "msg": "non-zero return code", "rc": 127, "start": "2024-11-21 11:00:18.989178", "stderr": "configure: error: in `/opt/sudo/sudo-1.8.31':\nconfigure: error: no acceptable C compiler found in $PATH\nSee `config.log' for more details\n/bin/sh: 2: make: not found\n/bin/sh: 3: make: not found", "stderr_lines": ["configure: error: in `/opt/sudo/sudo-1.8.31':", "configure: error: no acceptable C compiler found in $PATH", "See `config.log' for more details", "/bin/sh: 2: make: not found", "/bin/sh: 3: make: not found"], "stdout": "configure: Configuring Sudo version 1.8.31\nchecking for gcc... no\nchecking for cc... no\nchecking for cl.exe... no", "stdout_lines": ["configure: Configuring Sudo version 1.8.31", "checking for gcc... no", "checking for cc... no", "checking for cl.exe... no"]}

Reply:
yeah the installation of the sudo is failing right now, I tried a bunch of ways around it but its just not working